packages/core/src/deployment.ts

```diff
@@ -73,11 +74,15 @@ export async function waitAndValidateDeployment(provider: EthereumProvider, depl
   }
 
   debug('succeeded verifying deployment', txHash);
-  if (await hasCode(provider, address)) {
-    return;
+  while (!(await hasCode(provider, address))) {
+    const startTime = Date.now();
+    const elapsedTime = Date.now() - startTime;
+    // We only poll for code if we are in the context of a tx
```

Comment:
I don't understand this comment. What do you mean by being in the context of a tx?

Reply:
I understand that, if there is no txHash, it means we are just checking an address, and not a transaction that was just run. If so, it doesn't make sense to wait for getCode to resolve, since the tx that deployed that code should have landed long ago.

I may be mistaken though! Let me know if so and I can change this condition.